### PR TITLE
Fix most esql.query request types

### PR DESCRIPTION
**Pull request by pquentin**

### CODE REVIEW

File: specification/esql/query/QueryRequest.ts [hunk 1]
```diff
@@ -51,12 +57,17 @@ export interface Request extends RequestBase {
...
+     * @doc_id esql-returning-localized-results
+     */
     locale?: string
     /**
      * To avoid any attempts of hacking or code injection, extract the values in a separate list of parameters. Use question mark placeholders (?) in the query string for each of the parameters.
      * @doc_id esql-query-params
      */
-    params?: Array<ScalarValue>
+    params?: Array<EsqlParam>
+    profile?: boolean
```
Comment (pquentin) [inline specification/esql/query/QueryRequest.ts:70]: This one is undocumented but used in the YAML tests, which is why I'm adding without details.

**Comment by pquentin** [comment]
With `make validate`, I get `✔ 174 out of 174 test request cases are passing.`. I'll need to understand what is going on.

**Comment by pquentin** [comment]
So I get `224/227` now (both locally and in CI thanks to #2690!), and all the remaining failures are related to `LOOKUP` and its new `tables` parameter that I'd like to tackle later.

File: specification/esql/query/QueryRequest.ts [hunk 2]
```diff
@@ -51,12 +57,17 @@ export interface Request extends RequestBase {
...
+     * Returns results (especially dates) formatted per the conventions of the locale.
+     * @doc_id esql-returning-localized-results
+     */
     locale?: string
     /**
      * To avoid any attempts of hacking or code injection, extract the values in a separate list of parameters. Use question mark placeholders (?) in the query string for each of the parameters.
      * @doc_id esql-query-params
      */
-    params?: Array<ScalarValue>
+    params?: Array<UserDefinedValue>
```
Comment (pquentin) [inline specification/esql/query/QueryRequest.ts:69]: I'm not sure about this. It allows values like `{"foo": "bar"}`, which sounds difficult to type?

Comment (Anaethelion) [reply specification/esql/query/QueryRequest.ts:69]: We should go for `Array<ScalarValue | UserDefinedValue>`, which can be simplified to `Array<FieldValue>`.

Comment (pquentin) [reply specification/esql/query/QueryRequest.ts]: Thanks, done!

**Review by l-trotta — APPROVED** [review]

**Pull request merged**